frontend/app/route-helpers/status-route-helpers.server.ts

```diff
@@ -0,0 +1,156 @@
+import type { Session } from '@remix-run/node';
+import { redirectDocument } from '@remix-run/node';
+import type { Params } from '@remix-run/react';
+
+import { UTCDate } from '@date-fns/utc';
+import { differenceInMinutes } from 'date-fns';
+import { omit } from 'moderndash';
+import { z } from 'zod';
+
+import type { ContextualAlertType } from '~/utils/application-code-utils.server';
+import { getLocaleFromParams } from '~/utils/locale-utils.server';
+import { getLogger } from '~/utils/logging.server';
+import { getCdcpWebsiteApplyUrl } from '~/utils/url-utils.server';
+
+const log = getLogger('apply-route-helpers.server');
+
+export interface StatusState {
+  readonly lastUpdatedOn: string;
+  readonly id: string;
+  readonly statusCheckResult?: {
+    alertType: ContextualAlertType;
+    name?: string;
+    id?: string;
+    statusId?: string | null;
+  };
+}
+
+/**
+ * Schema for validating UUID.
+ */
+const idSchema = z.string().uuid();
+
+/**
+ * Gets the session name.
+ * @param id - The ID.
+ * @returns The session name.
+ */
+function getSessionName(id: string) {
+  return `status-flow-${idSchema.parse(id)}`;
+}
+
+interface LoadStateArgs {
+  params: Params;
+  session: Session;
+}
+
+/**
+ * Loads state.
+ * @param args - The arguments.
+ * @returns The loaded state.
+ */
+export function loadStatusState({ params, session }: LoadStateArgs) {
+  const locale = getLocaleFromParams(params);
+  const cdcpWebsiteApplyUrl = getCdcpWebsiteApplyUrl(locale);
+
+  const parsedId = idSchema.safeParse(params.id);
+
+  if (!parsedId.success) {
+    log.warn('Invalid "id" param format; redirecting to [%s]; id: [%s], sessionId: [%s]', cdcpWebsiteApplyUrl, params.id, session.id);
+    throw redirectDocument(cdcpWebsiteApplyUrl);
+  }
+
+  const sessionName = getSessionName(parsedId.data);
+
+  if (!session.has(sessionName)) {
+    log.warn('Status session state has not been found; redirecting to [%s]; sessionName: [%s], sessionId: [%s]', cdcpWebsiteApplyUrl, sessionName, session.id);
+    throw redirectDocument(cdcpWebsiteApplyUrl);
+  }
+
+  const state: StatusState = session.get(sessionName);
+
+  // Checks if the elapsed time since the last update exceeds 20 minutes,
+  // and performs necessary actions if it does.
+  const lastUpdatedOn = new UTCDate(state.lastUpdatedOn);
+  const now = new UTCDate();
+
+  if (differenceInMinutes(now, lastUpdatedOn) >= 20) {
```

Comment:
I understand this was taken from the existing `apply-route-helpers`, but we should consider allowing this 20 minutes to be configurable (in a future PR). 

Since the session dialog timeout is configurable, it could be changed to something greater than 20 minutes, which would mean the status session state could be wiped without any warning to the user.

Feel free to resolve - just wanted to point this out.